// lodash/lodash.js

		const _ = {
			get: function(object, path, defaultValue) {
				const keys = path.split('.');
				let result = object;
				for (const key of keys) {
					if (result && typeof result === 'object' && key in result) {
						result = result[key];
					} else {
						return defaultValue;
					}
				}
				return result;
			},
			set: function(object, path, value) {
				const keys = path.split('.');
				let current = object;
				for (let i = 0; i < keys.length - 1; i++) {
					const key = keys[i];
					if (!(key in current) || typeof current[key] !== 'object') {
						current[key] = {};
					}
					current = current[key];
				}
				current[keys[keys.length - 1]] = value;
				return object;
			},
			clone: function(obj) {
				return JSON.parse(JSON.stringify(obj));
			},
			merge: function(target, ...sources) {
				return Object.assign(target, ...sources);
			},
			isEmpty: function(value) {
				return value == null || (typeof value === 'object' && Object.keys(value).length === 0);
			},
			isArray: Array.isArray,
			isObject: function(value) {
				return value != null && typeof value === 'object' && !Array.isArray(value);
			},
			map: function(collection, iteratee) {
				if (Array.isArray(collection)) {
					return collection.map(iteratee);
				}
				return Object.keys(collection).map(key => iteratee(collection[key], key));
			},
			filter: function(collection, predicate) {
				if (Array.isArray(collection)) {
					return collection.filter(predicate);
				}
				const result = {};
				Object.keys(collection).forEach(key => {
					if (predicate(collection[key], key)) {
						result[key] = collection[key];
					}
				});
				return result;
			}
		};

		module.exports = _;
